util/changes.go

```diff
@@ -0,0 +1,42 @@
+package util
+
+import (
+	"reflect"
+)
+
+// RecursivelySetUndefinedFields sets all fields that are not set in structToSet to the value of the corresponding field in structToDefaultFrom.
+// It takes two reflect.Values, the first is the struct to set the fields on, and the second is the struct to get the default values from.
+func RecursivelySetUndefinedFields(structToSet, structToDefaultFrom reflect.Value) {
+	if structToSet.Kind() == reflect.Ptr {
```

Comment:
Can you explain this logic? It looks to me like the issue would be fixed if you just didn't pass in pointers, but I'm not sure.

Reply:
I think this should also have a comment, since reflect is confusing and we won't know looking back later why you've done this.

Reply:
Since this function is recursive we don't have any say if a pointer gets passed in as one of the nested fields may have a pointer. This casing handles that case.